builtin/credential/approle/path_login.go

```diff
@@ -33,7 +33,11 @@ func pathLogin(b *backend) *framework.Path {
 // Returns the Auth object indicating the authentication and authorization information
 // if the credentials provided are validated by the backend.
 func (b *backend) pathLoginUpdate(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	role, roleName, metadata, err := b.validateCredentials(req, data)
+	if req.Connection == nil || req.Connection.RemoteAddr == "" {
+		return nil, fmt.Errorf("failed to get connection information")
```

Comment:
OK, so -- I looked and it *is* possible for `req.Connection.RemoteAddr` to be an empty string. This only happens in an error case -- if a `net.SplitHostPort` call off of the `http.Request.RemoteAddr` errors out. So should this ever happen? No. However, the rest of the codebase allows it. So it may be better to have just the first part of the check (req.Connection being nil or not) but not the second. But I'm fine with you leaving both in.